packages/checkout-storefront/src/sections/SignIn/useSignInForm.ts

```diff
@@ -0,0 +1,55 @@
+import { AccountErrorCode } from "@/checkout-storefront/graphql";
+import { useErrorMessages } from "@/checkout-storefront/hooks/useErrorMessages";
+import { ApiError } from "@/checkout-storefront/hooks/useGetParsedErrors/types";
+import { useGetParsedErrors } from "@/checkout-storefront/hooks/useGetParsedErrors";
+import { useForm } from "@/checkout-storefront/hooks/useForm";
+import { useFormSubmit } from "@/checkout-storefront/hooks/useFormSubmit";
+import { useAuth } from "@saleor/sdk";
+import { object, string } from "yup";
+
+interface SignInFormData {
+  email: string;
+  password: string;
+}
+
+export const useSignInForm = ({ onSuccess }: { onSuccess: () => void }) => {
+  const { login } = useAuth();
+  const { getFormErrorsFromApiErrors } = useGetParsedErrors<SignInFormData>();
+  const { errorMessages } = useErrorMessages();
+
+  const validationSchema = object({
+    password: string().required(errorMessages.required),
+  });
+
+  const defaultFormData: SignInFormData = {
+    email: "",
+    password: "",
+  };
+
+  // @ts-ignore because login comes from the sdk which is no longer
```

Comment:
Maybe `ts-expect-error` would be better? Also do we have an issue to update that?

Reply:
Sure, makes sense. will change. We have an [issue to get rid of SDK ](https://github.com/orgs/saleor/projects/5/views/1?filterQuery=sdk&pane=issue&itemId=15805618)